pulsar-client/src/main/java/org/apache/pulsar/client/impl/ClientCnx.java

```diff
@@ -241,6 +244,7 @@ public void channelActive(ChannelHandlerContext ctx) throws Exception {
         super.channelActive(ctx);
         this.localAddress = ctx.channel().localAddress();
         this.remoteAddress = ctx.channel().remoteAddress();
+        internalPinnedExecutor = ctx.channel().eventLoop();
```

Comment:
Why not use the internal thread pool of the Pulsar client?

Here looks like switching to another IO thread. Either don't introduce a thread switch here or switch to the client's internal thread to reduce the IO thread burden.